src/Symfony/Bundle/SecurityBundle/DependencyInjection/Compiler/RegisterCsrfFeaturesPass.php

```diff
@@ -27,29 +27,25 @@ class RegisterCsrfFeaturesPass implements CompilerPassInterface
 {
     public function process(ContainerBuilder $container)
     {
-        if (!$container->has('security.csrf.token_storage')) {
-            return;
-        }
-
         $this->registerCsrfProtectionListener($container);
         $this->registerLogoutHandler($container);
     }
 
     private function registerCsrfProtectionListener(ContainerBuilder $container)
     {
-        if (!$container->has('security.authenticator.manager')) {
+        if (!$container->has('security.csrf.token_manager')) {
```

Comment:
I think we should keep the check for `security.authenticator.manager` here as well, because that's checking if the new system is used. If it isn't used, there is no need to register the listener service. (it won't be automatically removed IIRC, as it's registered as a listener)

Reply:
You're right. I have done it.